lib/decorate.js

```diff
@@ -1,44 +1,47 @@
 'use strict'
 
-function decorate (name, fn, dependencies) {
-  if (checkExistence(this, name)) {
+function decorate (instance, name, fn, dependencies) {
+  if (checkExistence(instance, name)) {
     throw new Error(`The decorator '${name}' has been already added!`)
```

Comment:
Let's go ahead and correct this grammar at the same time. Should be "The decorator '${name}' has already been added!'".